config/application.rb

```diff
@@ -29,7 +30,11 @@ class Application < Rails::Application
     config.autoload_lib(ignore: %w(assets tasks))
 
     config.view_component.default_preview_layout = "component_preview"
-    
+
+    # allow model-viewer web component to be used in views (and not stripped)
+    config.action_view.sanitized_allowed_tags = Rails::Html::SafeListSanitizer.allowed_tags.add('model-viewer')
```

Comment:
I needed to add `model-viewer` and its attributes to the list of things that Rails will allow in templates.

Reply:
We aren't using the sanitize helper, so I'm not sure this is doing anything.   Rails does not sanitize templates. You can put anything in them, even if it's not valid.

Reply:
At least in development the `<model-viewer>` component disappeared entirely from the delivered HTML until I added that. I can double check that's the case though.

Reply:
Also, FWIW, Rails uses the sanitize helper.

Reply:
Actually, you're right @jcoyne. I don't know what I was doing wrong, but after removing this config it still works. Thanks for catching this!